adagrams/game.py

```diff
@@ -1,11 +1,181 @@
\ No newline at end of file
+    highest_score = 0
+    highest_word = ""
+    for word in word_list:
+        score_of_word = score_word(word)
+        if score_of_word > highest_score:
+            highest_word, highest_score = replace_highest_values(highest_word, highest_score, word, score_of_word)
+        elif score_of_word == highest_score:
+            if len(word) == 10 and len(highest_word) !=10:
+                highest_word, highest_score = replace_highest_values(highest_word, highest_score, word, score_of_word)
+            elif len(word) <len(highest_word) and len(highest_word) != 10:
+                highest_word, highest_score = replace_highest_values(highest_word, highest_score, word, score_of_word)
+        print(score_of_word)
+    return tuple([highest_word, highest_score])
+
+
+
+
+
+# def get_highest_word_score_working(word_list):
+#     highest_score = 0
+#     highest_word = ""
+#     for word in word_list:
+#         score_of_word = score_word(word)
+#         if score_of_word > highest_score:
+#             highest_score = score_of_word
+#             highest_word = word
+#         elif score_of_word == highest_score:
+#             if len(word) == 10 and len(highest_word) !=10:
+#                 highest_score = score_of_word
+#                 highest_word = word
+#             elif len(word) <len(highest_word) and len(highest_word) != 10:
+#                 highest_score = score_of_word
+#                 highest_word = word
+#         print(score_of_word)
+#     return tuple([highest_word, highest_score])
+
+
+# A less efficient, but more awesome solution: 
+"""
+def get_highest_word_score(word_list):
+    list_scores = []
+
+    # checking for empty word_list input
+    if not word_list:
+        return None
+    
+    for word in word_list:
+        word_score_dict = {}
+        score_of_word = int(score_word(word))
+        # populating new dict with words and score
+        word_score_dict["word"] = word
+        word_score_dict["score"] = score_of_word
+        word_score_dict["len"] = len(word_score_dict["word"])
+        list_scores.append(word_score_dict)
+                 
+    # print(list_scores)    
+    highest_word_element = max(list_scores, key = lambda word: word["score"])
+    highest_score = highest_word_element["score"]
+
+    highest_score_list = []
+
+    for word in list_scores:
+        if word['score'] == highest_score:
+            highest_score_list.append(word)
+    
+    if len(highest_score_list) == 1:
+        return tuple([highest_score_list[0]['word'], highest_score_list[0]['score']])
+        
+    else:
+        for word in highest_score_list:
+            if word['len'] == 10:
+                print(tuple([word['word'], word['score']])) 
+                return tuple([word['word'], word['score']]) 
+        smallest_len = min(highest_score_list, key = lambda word: word["len"])
+        return tuple([smallest_len['word'], smallest_len['score']]) # need to double-check tuple return
```

Comment:
I agree that this is definitely a less efficient way to achieve what you had before, but it is certainly more awesome! Thank y'all for sharing that here. It shows me that y'all are thinking outside of the box and testing different ways of doing things which I can absolutely appreciate!